Lib/test/test_fractions.py

```diff
@@ -1530,6 +1530,8 @@ def test_format_f_presentation_type(self):
             (F(51, 1000), '.1f', '0.1'),
             (F(149, 1000), '.1f', '0.1'),
             (F(151, 1000), '.1f', '0.2'),
+            (F(1, 3), '.016f', '0.3333333333333333'),  # issue gh-130662
```

Comment:
Nitpicks:

`'.02f'` is simpler, and would not require to count digits in the result. We need also a 2-digit precision like in `'.010f'` to distinguish octals from decimals. Using `F(22, 7)` instead of `F(1, 3)` will produce a result in which it is easier to count digits.